clinica/pipelines/t1_linear/t1_linear_cli.py

```diff
@@ -0,0 +1,92 @@
+# coding: utf8
+
+"""T1 Linear - Clinica Command Line Interface.
+This file has been generated automatically by the `clinica generate template`
+command line tool. See here for more details:
+http://clinica.run/doc/InteractingWithClinica/
+"""
+
+
+import clinica.engine as ce
+
+
+class T1LinearCLI(ce.CmdParser):
+
+    def define_name(self):
+        """Define the sub-command name to run this pipeline.
+        """
+
+        self._name = 't1-linear'
+
+    def define_description(self):
+        """Define a description of this pipeline.
+        """
+        self._description = ('Brief description:\n'
+                             'http://clinica.run/doc/Pipelines/T1Linear/')
+
+    def define_options(self):
+        """Define the sub-command arguments
+        """
+        from clinica.engine.cmdparser import PIPELINE_CATEGORIES
+
+        # Clinica compulsory arguments (e.g. BIDS, CAPS, group_id...)
+        clinica_comp = self._args.add_argument_group(PIPELINE_CATEGORIES['CLINICA_COMPULSORY'])
+        clinica_comp.add_argument("bids_directory",
+                                  help='Path to the BIDS directory.')
+        clinica_comp.add_argument("caps_directory",
+                                  help='Path to the CAPS directory.')
+
+        # group_id can be used by certain pipelines when some operations are performed at the group level
+        # (for example, generation of a template in pipeline t1-volume)
+        # clinica_comp.add_argument("group_id",
+        #                          help='User-defined identifier for the provided group of subjects.')
+
+        # Clinica standard arguments
+        clinica_opt = self._args.add_argument_group(PIPELINE_CATEGORIES['CLINICA_OPTIONAL'])
+        clinica_opt.add_argument("-tsv", "--subjects_sessions_tsv",
+                                 help='TSV file containing a list of subjects with their sessions.')
+        clinica_opt.add_argument("-wd", "--working_directory",
+                                 help='Temporary directory to store pipelines intermediate results.')
+        clinica_opt.add_argument("-np", "--n_procs",
+                                 metavar=('N'), type=int,
+                                 help='Number of cores used to run in parallel.')
+
+        # Add your own pipeline command line arguments here to be used in the
+        # method below. Example below:
+        optional = self._args.add_argument_group(PIPELINE_CATEGORIES['OPTIONAL'])
+        optional.add_argument("-rt", "--ref_template",
+                              help='Reference template for registration.')
+
```

Comment:
Quick question: why the reference template needs to be provided by the user? I saw you put a downloading mechanism for this template.

Currenly, for any atlas, either it is embedded in Clinica or it embedded via neuroimaging software packages (e.g. JHU* for FSL, Neuromorphometrics & cie for CAT12).

Reply:
> Quick question: why the reference template needs to be provided by the user? I saw you put a downloading mechanism for this template.

I just kept it as an optional argument (e.g., if user wants to change the template). I don't known if it's really useful. I will remove it in the next commit.

> Currenly, for any atlas, either it is embedded in Clinica or it embedded via neuroimaging software packages (e.g. JHU* for FSL, Neuromorphometrics & cie for CAT12).

I don't want to distribute the templates with the package (in order to preserve a light package). They made almost 50Mb. I'm trying to add this "downloading mechanism" as a method of the module `clinica/utils/inputs.py`. It can be useful in other cases.